array_methods_lab.js

```diff
@@ -0,0 +1,371 @@
+// For spacing.
+const lineBreak = '--------------------------';
+
+// 1. Given an array of numbers, write a function that returns a new array where every element is incremented by 10.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: [11,12,13,14,15]
+// ```
+let arr = [1, 2, 3, 4, 5];
+
+const makeNewArr = (arr) => {
+  let incrementedByTen = arr.map((value => {
+    return value + 10;
+  }));
+    return incrementedByTen;
+}
+console.log(makeNewArr(arr));
+
+console.log (lineBreak);
+//
+// 2. Given an array, return a new array where every non-string element is converted to an empty string.
+//
+// ```
+// Input: ["a", 123, "b", "c", {name: "cat"}]
+// Output: ["a", "", "b", "c", ""]
+// ```
+let mixArr = ["a", 123, "b", "c", {name: "cat"}]
+let string = typeof 'sjdnfaiw';
+
+const convertToEmpty = (arr) => {
+  let mapNonString = arr.map ((value => {
+    if (typeof value !== string) {
+      return '';
+    } else {
+      return value;
+    }
+  }));
+  return mapNonString;
+}
+console.log (convertToEmpty(mixArr));
+
+console.log (lineBreak);
+//
+// 3. Write a function that returns only the even elements in an array of numbers.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: [2,4]
+// ```
+let numArr = [1, 2, 3, 4, 5];
+
+const givesEvenNumbers = (arr) => {
+  let onlyEvenElements = arr.filter((num => {
+    if (num % 2 === 0) {
+      return num;
+    }
+  }));
+  return onlyEvenElements;
+};
+
+console.log (givesEvenNumbers(numArr));
+
+console.log (lineBreak);
+//
+// 4. Write a function that checks if all elements in the array are the same.
+//
+// ```
+// Input: [1,1,1,1,1]
+// Output: true
+//
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+let copyArr = [1, 1, 1, 1, 1];
+let copyArr2 = [1, 1, 1, 1, 2];
+
+const checksAllAreSame = (arr) => {
+  let checkArray = arr.every (value => {
+    if (value === arr[value]) {
+      return true;
+    } else {
+      return false;
+    }
+  });
+  return checkArray;
+}
+
+console.log(checksAllAreSame(copyArr));
+
```

Comment:
Great , to simplify your code. Remember that every returns a boolean so, if you return your condition you'll be fine.